### PR TITLE
blog: how to build an application with dashboard 2.0

### LINKED ISSUE #1951: Blog art: How to build an application with Node-RED Dashboard 2.0
### Related Issue
https://github.com/FlowFuse/customer/issues/157

### Description

**Considerations:**
- Blog tile : How to build an application with Node-RED Dashboard 2.0

**content reference:**
https://github.com/FlowFuse/website/pull/1950

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Related Issue(s)

closes #1951
closes https://github.com/FlowFuse/customer/issues/157

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+In this guide, we are going to build a simple, secure task management application that will allow users to submit and view their own tasks, making the dashboard more personalized. Here's an overview of the steps:
+
+1. Accessing User Data: We have already enabled the application to access user data by installing the FlowFuse User Addon and enabling user authentication.
+2. Building a Form to Submit Tasks: We will create a form using Node-RED Dashboard 2.0 to allow users to input task details such as title, description, due date, etc.
+3. Storing Tasks in the Global Context: Upon task submission, we'll store this data alongside the associated user object in the global context of Node-RED.
+4. Displaying Task Submission Confirmation: Additionally, after a successful task submission, we'll display a notification.
+5. Retrieving and Filtering Tasks: We will retrieve the stored tasks from the global context and filter them based on the logged in user. This ensures that users only see their own tasks.
+6. Building a Customized Table: Using ui-template and Vuetify components, we will design a customized table on the dashboard to display the filtered tasks in a visually appealing format.
```
[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md]: I removed this part from blog

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+date: 2024-04-11
+authors: ["sumit-shinde"]
+image: /blog/2024/04/images/build-application-dashboard-2.png
+tags:
+    - posts
+    - node-red
+    - dashboard
+---
+
+We started developing Dashboard 2.0 a few months ago to replace the outdated Node-RED Dashboard 1.0. Initially, our goal was to replicate the old features and achieve feature parity. During the development Dashboard 2.0 has surpassed the deprecated Node-RED Dashboard, and it's now possible to build advanced applications. In this guide we'll build a Todo application to guide you through the features and explain how you can build rich and dynamic applications too.
```
Comment (ZJvandeWeg) [inline src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md:14]: Reading it again I think this post is about showing what's possible with D2.0. The current intro is a bit unguided, like it doesn't know why we're talking about this topic IMO.

```suggestion
In this guide we'll build a Todo application to guide you through the features and explain how you can build rich and dynamic applications too. It shows many of the features that make Dashboard 2.0 great, and why you should use it over the deprecated node-red-dashboard.
```

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+<!--more-->
+
+
+If you're new to Dashboard 2.0, refer to our blog post [Getting Started with Dashboard 2.0](/blog/2024/03/dashboard-getting-started/) to install and get things started.
+
+## Installing Flowfuse user addon
```
Comment (ZJvandeWeg) [inline src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md:21]: @joepavitt I think in FlowFuse it should come preinstalled somehow.

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+!["Screenshot of ui widget form configuration"](./images/building-an-application-with-dashboard-2-task-submission-form.png "Screenshot of ui widget form configuration"){data-zoomable}
+
+### Storing Tasks in the Global Context
+
+For this guide, we are storing our Tasks in Node-RED global context but storing them  in a database will make it easy to manage your task data.
+
+1. Drag a **function** node onto the canvas
+2. Paste the below code in the **function** node.
+
+```
```
Comment (ZJvandeWeg) [inline src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md:53]:
```suggestion
```javascript
```

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+Now that we can store tasks along with the user details of who submitted them, we need to retrieve and filter them based on users, ensuring that users can only see their tasks only and not others.
+
+1. Drag a **ui-event** widget onto the canvas and select **ui-base** for it. The **ui-event** will enable us to display updated tasks on the table without the need for polling, as it triggers when the page reloads or changes.
+2. Drag a **change** node onto the canvas and set `msg.payload` to `global.tasks`.
+
+!["Screenshot of the change setting retriving global context and setting to msg.payload"](./images/building-an-application-with-dashboard-change-node-retriving-global-context-data.png "Screenshot of the change setting retriving global context and setting to msg.payload"){data-zoomable}
+
+3. Drag a **function** node onto the canvas and paste the below code into it.
+
+```
```
Comment (ZJvandeWeg) [inline src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md:96]:
```suggestion
```javascript
```

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+In this section, we will build an interactive table using **ui-template** and [vuetify component](https://vuetifyjs.com/en/components/all/). Vuetify offers a wide range of components, all of which are compatible with our Node-RED Dashboard 2.0's ui-template widget. You can easily use them by just simply copying and pasting them into the **ui-template** widget.
+
+1. Drag an **ui-template** widget onto the canvas
+2. Create a new **ui-page** and **ui-group** for it. Below, I have provided a screenshot of the "new task" page configurations. Again You can replicate it if you want to align with the layout described in this guide, otherwise, it is optional.
+
+!["Screenshot displaying ui-template widget with code for building table for displaying task"](./images/building-an-application-with-dashboard-template-widget.png "Screenshot displaying ui-template widget with code for building table for displaying task"){data-zoomable}
+
+3. Paste the below code into the widget, If you're new to Vue.js, rest assured I've included helpful comments for clarity.
+
+```
```
Comment (ZJvandeWeg) [inline src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md:129]:
```suggestion
```html
```

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md (hunk 3 above)
Comment (joepavitt) [reply src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md:21]: FYA @MarianRaphael @hardillb

File: src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md [hunk 7]
```diff
@@ -0,0 +1,224 @@
+---
+title: How to Build An Application With Node-RED Dashboard 2.0
+subtitle: A step-by-step guide to building a personalized, secure, and fully functional application with Dashboard 2.0.
+description: Learn to build custom applications effortlessly with Node-RED Dashboard 2.0. This step-by-step guide walks you through building a personalized, secure, and fully functional application.
+date: 2024-04-11
```
Comment (ZJvandeWeg) [inline src/blog/2024/04/how-to-build-an-application-with-node-red-dashboard-2.md:5]:
```suggestion
date: 2024-04-25
```

**Review by ZJvandeWeg — APPROVED** [review]

**Pull request merged**